### PR TITLE
Rename ndp to bolt

### PR DESCRIPTION
**Pull request by srbaker**

### CODE REVIEW

File: community/bolt/README.md [hunk 1]
```diff
@@ -1,7 +1,6 @@
-# Neo4j Data Protocol
+# Bolt
```
Comment (technige) [inline community/bolt/README.md]: How about:

`Bolt: The Neo4j Binary Protocol`

File: community/bolt/kernelextension/pom.xml [hunk 2]
```diff
@@ -16,10 +16,10 @@
     <licensing.prepend.text>notice-gpl-prefix.txt</licensing.prepend.text>
   </properties>
 
-  <artifactId>neo4j-ndp-kernelextension</artifactId>
+  <artifactId>neo4j-bolt-kernelextension</artifactId>
 
   <packaging>jar</packaging>
-  <name>Neo4j - NDP Kernel Extension</name>
+  <name>Neo4j - Bolt Kernel Extension</name>
   <description>Exposes the Neo4j Data Protocol server as a kernel extension.</description>
```
Comment (technige) [inline community/bolt/kernelextension/pom.xml]: ...the Neo4j Bolt Protocol server...

File: community/bolt/kernelextension/src/main/java/org/neo4j/ext/BoltKernelExtension.java [hunk 3]
```diff
@@ -134,7 +134,7 @@ public SocketProtocol apply( Channel channel )
             life.add( new NettyServer( scheduler.threadFactory( gapNetworkIO ), asList(
```
Comment (technige) [inline community/bolt/kernelextension/src/main/java/org/neo4j/ext/BoltKernelExtension.java]: Wondering if we need to replace instances of "gap" as well as "ndp".

File: community/bolt/v1-docs/src/docs/dev/examples.asciidoc [hunk 4]
```diff
@@ -1,4 +1,4 @@
-[[ndp-examples]]
+[[bolt-examples]]
 == Examples
 
 This section contains concrete examples showing how to perform tasks using the full protocol stack.
```
Comment (srbaker) [inline community/bolt/v1-docs/src/docs/dev/examples.asciidoc]: "The full Bolt protocol stack."

File: community/bolt/v1-docs/src/docs/dev/index.asciidoc [hunk 5]
```diff
@@ -1,4 +1,4 @@
-[[ndp]]
+[[bolt]]
 = Neo4j Data Protocol, Version 1
```
Comment (srbaker) [inline community/bolt/v1-docs/src/docs/dev/index.asciidoc:3]: "Neo4j Bolt Protocol"

File: community/bolt/v1-docs/src/docs/dev/messaging.asciidoc [hunk 6]
```diff
@@ -22,10 +22,10 @@ image:simple-exchange.png[]
 
 === Initialization
 
-Before a session can be used to <<ndp-message-structs-run,run>> queries, it must be <<ndp-message-structs-initialize,initialized>>.
-The `INITIALIZE` message message should be sent by the client as the first message it sends after <<ndp-handshake,negotiating protocol version>>.
+Before a session can be used to <<bolt-message-structs-run,run>> queries, it must be <<bolt-message-structs-initialize,initialized>>.
+The `INITIALIZE` message message should be sent by the client as the first message it sends after <<bolt-handshake,negotiating protocol version>>.
```
Comment (srbaker) [inline community/bolt/v1-docs/src/docs/dev/messaging.asciidoc]: s/message message/message/

File: community/bolt/v1-docs/src/docs/dev/transport.asciidoc [hunk 7]
```diff
@@ -1,4 +1,4 @@
-[[ndp-transport]]
+[[bolt-transport]]
 == Transport layer
```
Comment (srbaker) [inline community/bolt/v1-docs/src/docs/dev/transport.asciidoc]: "Bolt transport layer"

File: community/bolt/v1-docs/src/docs/dev/transport.asciidoc [hunk 8]
```diff
@@ -31,7 +31,7 @@ IMPORTANT: If TLS is enabled, and no certificate has been specified, Neo4j will
  you do not verify the certificate it is very simple to bypass the encryption. You should either ensure you have a
  valid signed certificate installed with Neo4j or that your implements +trust on first use+.
 
-[[ndp-handshake]]
+[[bolt-handshake]]
 === Handshake
 
 After connecting, a handshake takes place to establish which protocol version should be used for that connection.
```
Comment (srbaker) [inline community/bolt/v1-docs/src/docs/dev/transport.asciidoc]: "Bolt protocol version"

File: community/bolt/v1-docs/src/test/java/org/neo4j/bolt/docs/v1/DocsRepository.java [hunk 9]
```diff
@@ -40,7 +40,7 @@
 public class DocsRepository
 {
     private final static String SEP = File.separator;
-    private final static File docsDir = findBackwards( "community" + SEP + "ndp" + SEP + "v1-docs" + SEP +
+    private final static File docsDir = findBackwards( "community" + SEP + "bolt" + SEP + "v1-docs" + SEP +
                                                        "src" + SEP + "docs" + SEP, 5 );
```
Comment (srbaker) [inline community/bolt/v1-docs/src/test/java/org/neo4j/bolt/docs/v1/DocsRepository.java:44]: This line of code is shit.

File: community/server/src/main/java/org/neo4j/server/configuration/ServerSettings.java [hunk 10]
```diff
@@ -177,19 +177,19 @@ private ThirdPartyJaxRsPackage createThirdPartyJaxRsPackage( String packageAndMo
...
-    @Description("Host and port for the Neo4j Data Protocol")
-    Setting<HostnamePort> ndp_socket_address = setting( "dbms.ndp.address", HOSTNAME_PORT, "localhost:7687" );
+    @Description("Host and port for Bolt")
+    Setting<HostnamePort> bolt_socket_address = setting( "dbms.bolt.address", HOSTNAME_PORT, "localhost:7687" );
 
     @Internal
-    @Description("Host and port for the Neo4j Data Protocol Websocket")
-    Setting<HostnamePort> ndp_ws_address = setting( "dbms.ndp.ws.address", HOSTNAME_PORT, "localhost:7688" );
+    @Description("Host and port for the Bolt Websocket")
+    Setting<HostnamePort> bolt_ws_address = setting( "dbms.bolt.ws.address", HOSTNAME_PORT, "localhost:7688" );
```
Comment (srbaker) [inline community/server/src/main/java/org/neo4j/server/configuration/ServerSettings.java:193]: let's spell this out: "websocket"

File: community/server/src/main/java/org/neo4j/server/configuration/ServerSettings.java [hunk 11]
```diff
@@ -177,19 +177,19 @@ private ThirdPartyJaxRsPackage createThirdPartyJaxRsPackage( String packageAndMo
...
-    @Description("Enable TLS for the Neo4j Data Protocol")
-    Setting<Boolean> ndp_tls_enabled = setting( "xx.ndp.tls.enabled", BOOLEAN, FALSE );
+    @Description("Enable TLS for Bolt")
+    Setting<Boolean> bolt_tls_enabled = setting( "xx.bolt.tls.enabled", BOOLEAN, FALSE );
 
     @Internal
-    @Description("Host and port for the Neo4j Data Protocol")
-    Setting<HostnamePort> ndp_socket_address = setting( "dbms.ndp.address", HOSTNAME_PORT, "localhost:7687" );
+    @Description("Host and port for Bolt")
+    Setting<HostnamePort> bolt_socket_address = setting( "dbms.bolt.address", HOSTNAME_PORT, "localhost:7687" );
```
Comment (srbaker) [inline community/server/src/main/java/org/neo4j/server/configuration/ServerSettings.java:189]: bolt.socket.address

File: community/server/src/main/java/org/neo4j/server/configuration/ServerSettings.java (hunk 10 above)
Comment (srbaker) [inline community/server/src/main/java/org/neo4j/server/configuration/ServerSettings.java:193]: bolt.websocket.address

File: community/server/src/main/java/org/neo4j/server/configuration/ServerSettings.java (hunk 11 above)
Comment (jakewins) [inline community/server/src/main/java/org/neo4j/server/configuration/ServerSettings.java:189]: +1

File: community/server/src/main/java/org/neo4j/server/configuration/ServerSettings.java (hunk 10 above)
Comment (jakewins) [inline community/server/src/main/java/org/neo4j/server/configuration/ServerSettings.java:193]: +1

**Comment by systay** [comment]
@srbaker CPR for this PR?

**Comment by davidegrohmann** [comment]
@srbaker @jakewins @nigelsmall rebase needed, what's the status of this PR?

**Comment by jakewins** [comment]
Rebasing and opening this in a new PR

**Pull request closed without merging**